### DISCUSSION TITLE
Patching a program.

### DISCUSSION THREAD

**Discussion by TheGithubJoshua** [eteran/edb-debugger#854] [Q&A]
Is it possible to patch a program in edb and if so how do you do it?

**Comment by eteran** [comment]
Good question, the answer is currently "yes" and "no" depending on what you mean.

Yes: you can alter the machine code and data of a running process to make it act differently

No: you cannot save these changes to disk so that all future executions of the program have the patch applied.

This is of course a completely reasonable feature to want. For small cases, it's actually pretty easy to implement too. However, once we run into situations where you need to change the size of sections or even add new sections) it becomes MUCH more complicated since this amounts to basically re-writing the ELF file to ensure everything ends up how it should be.
